src/main/java/servlets/BillingConfig.java

```diff
@@ -97,13 +89,19 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    // POST method --> overwrites the existing configuration or creates a new one
-    // currently assuming there is only one account --> add this to the form @Vincent?
-    // TODO: can the vendor ID be appended to the request as a query string @vincent?
-    String tempVendorID = "ALPHA";
-    Vendor newVendor = new Vendor(request, tempVendorID, 1);
-    JsonConverter jsonConverter = new JsonConverter();
     response.setContentType(CONTENT_TYPE_APPLICATION_JSON);
-    response.sendRedirect("/index.html");
+    try {
+      Vendor newVendor = new Vendor(request);
+      JsonConverter jsonConverter = new JsonConverter();
+      if (jsonConverter.updateFile(newVendor)) {
+        // Redirect only when the operation succeeded
+        response.getWriter().println(newVendor.getVendorID());
+        response.sendRedirect(REDIRECT_READFILE);
+      } else {
+        response.sendError(400, "Something went wrong when we tried to write your file.");
+      }
+    } catch(NumberFormatException e) {
+      response.sendError(400, "A NumberFormatException occurred.");
```

Comment:
Can we use a more informative error message here? Like "invalid vendor ID" or similar

Reply:
yes, but I'm not sure what the error should be. The updateFile method returns the boolean `file != null`, and I haven't found anything in testing that I can do to have the file actually return null. I think this error will only happen if the file can't be created

Reply:
resolved in #71